src/modules/accommodations/accommodations.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@nestjs/common';
+import { Accommodation } from './entities';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { SearchAccommodationParamsDto } from './dto/request';
+import { PaginatedResult } from './interfaces';
+
+@Injectable()
+export class AccommodationsService {
+  constructor(
+    @InjectRepository(Accommodation)
+    private readonly accommodationRepository: Repository<Accommodation>,
+  ) {}
+
+  async getManyAccommodations(
+    searchParams: SearchAccommodationParamsDto,
+  ): Promise<PaginatedResult<Accommodation>> {
+    const query =
+      this.accommodationRepository.createQueryBuilder('accommodation');
+
+    const page = searchParams.page || 1;
+    const limit = searchParams.limit || 10;
+    const skip = (page - 1) * limit;
+
+    if (searchParams.search) {
+      query.andWhere(
+        '(accommodation.name ILIKE :search OR accommodation.description ILIKE :search)',
+        { search: `%${searchParams.search}%` },
+      );
+    }
+
+    if (searchParams.category) {
+      query.innerJoinAndSelect(
+        'accommodation.category',
+        'category',
+        'category.id = :categoryId',
+        { categoryId: searchParams.category },
+      );
+    }
+
+    if (searchParams.location) {
+      query.andWhere('(accommodation.location ILIKE :location)', {
+        location: `%${searchParams.location}%`,
+      });
+    }
+
+    if (searchParams.fromDate && searchParams.toDate) {
+      query.andWhere(
```

Comment:
It won't check by creation date. We need another booking table for handling booking and availibility date. Just remove this part. It doesn't work now